lib/cell.rb

```diff
@@ -0,0 +1,41 @@
+class Cell
+  attr_reader :coordinate,
+              :ship
+
+  def initialize(coordinate)
+    @coordinate = coordinate
+    @ship = nil
+    @fire_status = false
+  end
+
+  def empty?
+    @ship == nil
+  end
+
+  def place_ship(ship)
+    @ship = ship
+  end
+
+  def fired_upon? 
+    @fire_status
+  end
+
+  def fire_upon
+    @ship&.hit 
+    @fire_status = true
+  end
+
+  def render(arg = false)
```

Comment:
This is looking great! One small enhancement -- if I'm just inspecting this class without much knowledge of what the board or game flow looks like, the argument name `arg` is pretty vague. Can you rename it to something that is a bit more illustrative of its purpose? Something like `show_ships`?